pilot/pkg/features/pilot.go

```diff
@@ -261,11 +261,10 @@ var (
 			"See https://godoc.org/k8s.io/client-go/rest#Config Burst",
 	).Get()
 
-	// IstiodService controls the istiod address - used for injection and as default value injected into pods
-	// if istiod is used. The name must be part of the DNS certificate served by pilot/istiod. The '.svc' is
-	// imposed by K8S - that's how the names for webhooks are defined, based on webhook service (which will be
-	// istio-pilot or istiod) plus namespace and .svc.
-	// The 15010 port is used with plain text, 15011 with Spiffe certs - we need a different port for DNS cert.
+	// IstiodService controls the istiod address - used for injection and as default value injected into pods.
+	// The name must be part of the DNS certificate served by istiod. The '.svc' is imposed by K8S - that's how
+	// the names for webhooks are defined, based on webhook service (which will be istiod) plus namespace and .svc.
+	// The 15010 port is used with plain text port 15012 is used with secure port.
```

Comment:
Can we set a default value ? It was off by default in 1.5 since it was a new value, now I think it should be on by default - using istiod.istio-system.svc:15012 value which is the prod default. Please add a TODO, probably separate PR would be better.

Alternatively - MeshConfig.discoveryAddress should be used.

Reply:
https://github.com/istio/istio/pull/23700 - Cleans up the feature flag and uses `MeshConfig.discoveryAddress`